core/promise/consumer_test.go

```diff
@@ -18,24 +18,157 @@
 	consumer := Consumer{}
+	response, err := consumer.Consume(0)
+	assert.Error(t, errUnsupportedRequest, err)
+	assert.Nil(t, response)
+}
 
+func TestConsumeBadSignature(t *testing.T) {
+	consumer := Consumer{}
 	signedPromise := &SignedPromise{Promise: Promise{}, IssuerSignature: "ProducerSignature"}
+	request := &Request{signedPromise}
+	response, err := consumer.Consume(request)
+	assert.Nil(t, err)
+	assert.Equal(t, &Response{
+		Success: false,
+		Message: errBadSignature.Error(),
+		Request: request,
+	}, response)
+}
+
+func TestConsumeUnknownBenefiter(t *testing.T) {
+	var request Request
+	err := json.Unmarshal(jsonRequest, &request)
+	assert.Nil(t, err)
+
+	consumer := Consumer{}
+	response, err := consumer.Consume(&request)
+	assert.Nil(t, err)
+	assert.Equal(t, &Response{
+		Success: false,
+		Message: errUnknownBenefiter.Error(),
+		Request: &request,
+	}, response)
+}
+
+func TestConsumeLowAmount(t *testing.T) {
+	var request Request
+	err := json.Unmarshal(jsonRequest, &request)
+	assert.Nil(t, err)
+
+	proposal := dto.ServiceProposal{
+		ProviderID:    "0x67c8afcfc5432cf1ce8d5e289aeacbffa1904f82",
+		PaymentMethod: fakePayment{999},
+	}
+	consumer := Consumer{proposal: proposal}
+	response, err := consumer.Consume(&request)
+	assert.Nil(t, err)
+	assert.Equal(t, &Response{
+		Success: false,
+		Message: errLowAmount.Error(),
+		Request: &request,
+	}, response)
+}
+
+func TestConsumeLowBalance(t *testing.T) {
+	var request Request
+	err := json.Unmarshal(jsonRequest, &request)
+	assert.Nil(t, err)
+
+	proposal := dto.ServiceProposal{
+		ProviderID:    "0x67c8afcfc5432cf1ce8d5e289aeacbffa1904f82",
+		PaymentMethod: fakePayment{1},
+	}
+	consumer := Consumer{proposal: proposal, etherClient: &fakeBlockchain{1}}
+	response, err := consumer.Consume(&request)
+	assert.Nil(t, err)
+	assert.Equal(t, &Response{
+		Success: false,
+		Message: errLowBalance.Error(),
+		Request: &request,
+	}, response)
+}
 
-	response, err := consumer.Consume(&Request{signedPromise})
+func TestConsume(t *testing.T) {
+	var request Request
+	err := json.Unmarshal(jsonRequest, &request)
+	assert.Nil(t, err)
+
+	proposal := dto.ServiceProposal{
+		ProviderID:    "0x67c8afcfc5432cf1ce8d5e289aeacbffa1904f82",
+		PaymentMethod: fakePayment{1},
+	}
+	consumer := Consumer{proposal: proposal, etherClient: &fakeBlockchain{99999}, storage: &fakeStorage{}}
+	response, err := consumer.Consume(&request)
 	assert.NoError(t, err)
-	assert.Equal(t, &Response{Success: true, Message: "Promise accepted"}, response)
+	assert.Equal(t, &Response{
+		Success: true,
+		Message: "Promise accepted",
+		Request: &request,
+	}, response)
+}
+
+type fakePayment struct {
+	amount uint64
+}
+
+func (fp fakePayment) GetPrice() money.Money {
+	return money.Money{Amount: fp.amount}
+}
+
+type fakeBlockchain struct {
+	balance int64
 }
+
+func (fb *fakeBlockchain) BalanceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (*big.Int, error) {
```

Comment:
If would have smaller interface, You would not need to mock all other functions `StorageAt`, `NonceAt `

Reply:
Fixed.